packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts

```diff
@@ -0,0 +1,436 @@
+    }
+    const mixedRouteIsAllV2 = (r: Omit<BaseRoute, 'input' | 'output'>) => {
+      return r.pools.every(SmartRouter.isV2Pool)
+    }
+
+    const mixedRouteIsAllStable = (r: Omit<BaseRoute, 'input' | 'output'>) => {
+      return r.pools.every(SmartRouter.isStablePool)
+    }
+
+    // similar to encodeMixedRouteSwap but more simplified where we just continue
+    // as if its a regular v2 or v3 trade
+    if (singleHop) {
+      if (mixedRouteIsAllV3(route)) {
+        addV3Swap(
+          planner,
+          {
+            ...trade,
+            routes: [route],
+            inputAmount,
+            outputAmount,
+          },
+          options,
+          routerMustCustody,
+          payerIsUser,
+          performAggregatedSlippageCheck,
+        )
+      } else if (mixedRouteIsAllV2(route)) {
+        addV2Swap(
+          planner,
+          {
+            ...trade,
+            routes: [route],
+            inputAmount,
+            outputAmount,
+          },
+          options,
+          routerMustCustody,
+          payerIsUser,
+          performAggregatedSlippageCheck,
+        )
+      } else if (mixedRouteIsAllStable(route)) {
+        addStableSwap(
+          planner,
+          {
+            ...trade,
+            routes: [route],
+            inputAmount,
+            outputAmount,
+          },
+          options,
+          routerMustCustody,
+          performAggregatedSlippageCheck,
+          payerIsUser,
+        )
+      } else {
+        throw new Error('Unsupported route to encode')
+      }
+    } else {
+      const sections = SmartRouter.partitionMixedRouteByProtocol(route)
+
+      const isLastSectionInRoute = (i: number) => {
+        return i === sections.length - 1
+      }
+
+      let outputToken
+      let inputToken = inputAmount.currency.wrapped
+
+      for (let i = 0; i < sections.length; i++) {
+        const section = sections[i]
+        /// Now, we get output of this section
+        outputToken = SmartRouter.getOutputOfPools(section, inputToken)
+        const newRoute = SmartRouter.buildBaseRoute([...section], inputToken, outputToken)
+        inputToken = outputToken.wrapped
+
+        const lastSectionInRoute = isLastSectionInRoute(i)
+        const recipientAddress = isLastSectionInRoute(i) ? recipient : ROUTER_AS_RECIPIENT
+
+        const inAmount = i === 0 ? amountIn : BigInt(2) ** BigInt(255)
+        const outAmount = !lastSectionInRoute ? 0n : amountOut
+
+        if (mixedRouteIsAllV3(newRoute as BaseRoute)) {
+          const pathStr = SmartRouter.encodeMixedRouteToPath(newRoute, !isExactIn)
+          if (isExactIn) {
+            planner.addCommand(CommandType.V3_SWAP_EXACT_IN, [
+              isLastSectionInRoute(i) ? recipient : (getPoolAddress(sections[i + 1][0]) as Address),
+              inAmount, // amountIn
+              outAmount, // amountOut
+              pathStr, // path
+              payerIsUser && i === 0, // payerIsUser
```

Comment:
let's define an inner variable for payerIsUser without writing the `&& i === 0` every time